Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -559,8 +556,10 @@ std::unique_ptr<MaybeDocument> Serializer::DecodeBatchGetDocumentsResponse(
         break;
 
       case google_firestore_v1beta1_BatchGetDocumentsResponse_transaction_tag:
-        // TODO(rsgowman)
-        abort();
+        // This field is ignored by the client sdk, but we still need to extract
+        // it.
+        reader->ReadString();
```

Comment:
Why do we need to read extra fields when we don't care about them? This seems like a *serious* problem: the server considers itself free to add fields we don't know about since we're not supposed to fail on unknown fields.

This specific implementation succeeds because it "knows" the transaction tag is of type string, but the server could just as easily send us an unknown field that contains anything, including a submessage of a type we didn't know at compile time.

If we need to take some affirmative action to ignore unknown fields I'd prefer we handle this in a way that's general so that we can apply it to all of our message read blocks.

Reply:
> Why do we need to read extra fields when we don't care about them?

Otherwise, our byte count will be off. We need to skip these bytes.

> This specific implementation succeeds because it "knows" the transaction tag is of type string, but the server could just as easily send us an unknown field that contains anything, including a submessage of a type we didn't know at compile time.

Yup.

> If we need to take some affirmative action to ignore unknown fields I'd prefer we handle this in a way that's general so that we can apply it to all of our message read blocks.

Already done (though obviously not in this PR.) The approach being:
a) Read tag
b) handle if known (ie what exists today)
c) otherwise inspect type and read (i.e. skip) the appropriate number of bytes.

For c, that means either read a primitive type (varint or fixed size) or a length delimited value (string, message, etc)

Reply:
I see. Makes sense.

In that case, I'd make some kind of marker or grouping of these intentionally ignored fields such that this explicit discard can ultimately be removed. That marker can even be a TODO to switch this to reader->IgnoreField or whatever once that's available.

In the fullness of time would you expect all known ignored fields to be grouped together (i.e. multiple `case` labels for the same body)?

Reply:
> In that case, I'd make some kind of marker or grouping of these intentionally ignored fields such that this explicit discard can ultimately be removed. That marker can even be a TODO to switch this to reader->IgnoreField or whatever once that's available.

Done.